### PR TITLE
Issue #423

### PR DESCRIPTION
**Pull request by lorban**
The bulk of stats rework is done, as well as the M&M code.

There's still some work left but this code already deserves reviews and comments.

### CODE REVIEW

File: 107/src/main/java/org/ehcache/jsr107/Eh107CacheStatisticsMXBean.java [hunk 1]
```diff
@@ -15,179 +15,211 @@
...
-
-  private final org.ehcache.statistics.CacheStatistics statistics;
-  private final ConcurrentMap<BulkOps, AtomicLong> bulkMethodCounts;
-  private final EnumMap<Stat, AtomicLong> baselines = new EnumMap<Stat, AtomicLong>(Stat.class);
-
-  Eh107CacheStatisticsMXBean(String cacheName, Eh107CacheManager cacheManager,
-      org.ehcache.statistics.CacheStatistics statistics) {
+public class Eh107CacheStatisticsMXBean extends Eh107MXBean implements javax.cache.management.CacheStatisticsMXBean {
+
+  private final Ehcache cache;
```
Comment (ljacomet) [inline 107/src/main/java/org/ehcache/jsr107/Eh107CacheStatisticsMXBean.java]: From looking at the code, we could get a hold of the `bulkMethodEntries` in the constructor and thus not hold a reference to the cache.

File: docs/src/docs/asciidoc/user/management.adoc [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+== Making use of the `ManagementRegistry`
+
+By default, a `ManagementRegistry` is automatically discovered and enabled, but can only be accessed by ehcache
+internal services. If you wish to make use of it, you should create your own instance and pass it to the
+cache manager builder as a service:
+
+[source,java]
+----
+ManagementRegistry managementRegistry = new DefaultManagementRegistry(); //// <1>
```
Comment (ljacomet) [inline docs/src/docs/asciidoc/user/management.adoc]: Can this be moved to be real sample code?
Ideally the code lives in a class in `org.ehcache.docs` package in one of the modules, which enables specific check style rules on it.
The 4 slashes are not needed, only two. Check other samples. The output includes the number as an image that is not copied over when you copy the code.

**Comment by ljacomet** [comment]
Don't break the TCK :smile:

**Comment by alexsnaps** [comment]
test this please

**Comment by alexsnaps** [comment]
Could you cleanup these commit message... Some squashing seems appropriate here. And have them include the issue number.

**Comment by lorban** [comment]
please test this

**Comment by lorban** [comment]
test this please

**Comment by lorban** [comment]
test this please

**Comment by ljacomet** [comment]
test this please

File: docs/src/docs/asciidoc/user/management.adoc [hunk 3]
```diff
@@ -0,0 +1,97 @@
+include::common.adoc[]
+
+= Management and Monitoring
```
Comment (ljacomet) [inline docs/src/docs/asciidoc/user/management.adoc]: Could this be referenced from menu.adoc?

**Comment by lorban** [comment]
test this please

**Pull request merged**